lib/Service/RoomService.php

```diff
@@ -392,4 +409,101 @@ public function verifyPassword(Room $room, string $password): array {
 			'url' => '',
 		];
 	}
+
+	public function setTimeToLive(Room $room, Participant $participant, int $ttl): void {
+		$event = new ChangeTtlEvent($room, $ttl);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_TIME_TO_LIVE, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('time_to_live', $update->createNamedParameter($ttl, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+		$jobList = Server::get(IJobList::class);
+		if ($ttl > 0) {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_enabled');
+			$jobList->add(ApplyTtl::class, ['room_id' => $room->getId()]);
+		} else {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_disabled');
+			$jobList->remove(ApplyTtl::class, ['room_id' => $room->getId()]);
+		}
```

Comment:
```suggestion
		if ($ttl > 0) {
			$jobList->add(ApplyTtl::class, ['room_id' => $room->getId()]);
			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_enabled');
		} else {
			$jobList->remove(ApplyTtl::class, ['room_id' => $room->getId()]);
			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_disabled');
		}
```

The background job is more important so let's do that first, so in case it fails we didn't add a chat message yet